packages/utils/src/schema/retrieveSchema.ts

```diff
@@ -395,35 +396,63 @@ export function stubExistingAdditionalProperties<
       // No need to stub, our schema already has the property
       return;
     }
-
-    let additionalProperties: S['additionalProperties'] = {};
-    if (typeof schema.additionalProperties !== 'boolean') {
-      if (REF_KEY in schema.additionalProperties!) {
-        additionalProperties = retrieveSchema<T, S, F>(
+    if (PATTERN_PROPERTIES_KEY in schema) {
+      const patternProperties = Object.keys(schema.patternProperties!)
+        .filter((pattern) => RegExp(pattern).test(key))
+        .reduce(
+          (obj, pattern) => {
+            obj[pattern] = schema.patternProperties![pattern];
+            return obj;
+          },
+          {} as NonNullable<S['patternProperties']>,
+        );
```

Comment:
This looks like a duplicate of the code at lines 551-559, consider drying things up with a helper method